packages/server/src/client-tokens/service.ts

```diff
@@ -0,0 +1,125 @@
+import { uuid } from '@botpress/messaging-base'
+import { CachingService, CryptoService, DatabaseService, ServerCache, Service } from '@botpress/messaging-engine'
+import crypto from 'crypto'
+import { validate as validateUuid, v4 as uuidv4 } from 'uuid'
+import yn from 'yn'
+import { ClientTokenTable } from './table'
+import { ClientToken } from './types'
+
+export const CLIENT_TOKEN_LENGTH = 66
+
+export class ClientTokenService extends Service {
+  private table: ClientTokenTable
+  private cacheById!: ServerCache<uuid, ClientToken>
+  private cacheTokens!: ServerCache<uuid, string>
+
+  constructor(
+    private db: DatabaseService,
+    private cryptoService: CryptoService,
+    private cachingService: CachingService
+  ) {
+    super()
+    this.table = new ClientTokenTable()
+  }
+
+  async setup() {
+    this.cacheById = await this.cachingService.newServerCache('cache_client_token_by_id')
+    this.cacheTokens = await this.cachingService.newServerCache('cache_client_token_raw')
+
+    await this.db.registerTable(this.table)
+  }
+
+  async generateToken(): Promise<string> {
+    return crypto.randomBytes(CLIENT_TOKEN_LENGTH).toString('base64')
+  }
+
+  async create(clientId: uuid, token: string, expiry: Date | undefined): Promise<ClientToken> {
+    const clientToken = {
+      id: uuidv4(),
+      clientId,
+      token: await this.cryptoService.hash(token),
+      expiry
+    }
+
+    await this.query().insert(this.serialize(clientToken))
+    this.cacheById.set(clientToken.id, clientToken)
+
+    return clientToken
+  }
+
+  async fetchById(id: uuid): Promise<ClientToken | undefined> {
+    const cached = this.cacheById.get(id)
+    if (cached) {
+      return cached
+    }
+
+    const rows = await this.query().where({ id })
+
+    if (rows?.length) {
+      const clientToken = this.deserialize(rows[0])
+      this.cacheById.set(id, clientToken)
+      return clientToken
+    } else {
+      return undefined
+    }
+  }
+
+  async verifyToken(clientId: string, rawToken: string): Promise<ClientToken | undefined> {
+    if (!rawToken?.length) {
+      return undefined
+    }
+
+    const [id, token] = rawToken.split('.')
```

Comment:
Can you put this separator in a constant so we can reuse it across services and in tests